cypress-io/cypress
# fix: Display correct major version in migration wizard

- Closes #25138 

### User facing changelog
* Fixes issue where incorrect Cypress version could be shown in the migration wizard
* Fixes issue where the Cypress migration wizard would fail to run in global mode on newer versions of Cypress

### Additional details
This resolves a couple hardcoded version references in messaging and also updates an edge-case compatibility check when a user is attempting to use Cypress in global mode to migrate a project with a different version of Cypress. The compatibility check only supported `^10.0.0` which would cause failures when trying to migrate v11/12 even if there wasn't really an incompatibility.

### Steps to test
1. Initialize a mock old Cypress project to migrate
```
mkdir myProject
cd myProject
echo '{}' > cypress.json
npm init
npm i -D cypress@11.2.0
```
3. Launch this branch in global mode and attempt to open the project. You should receive an error message warning of a version mismatch (global v12.1.0 is greater than local project 11.2.0)
4. Close and update mock project Cypress version `npm i -D cypress@12.1.0`
5. Re-launch this branch, you should no longer receive error message
6. Validate that Migration Wizard displays "Migrate to Cypress 12" title

### How has the user experience changed?
**Error Condition**
_Before_
![Screen Shot 2022-12-14 at 9 26 37 AM](https://user-images.githubusercontent.com/11482842/207637928-a8a74206-8f96-4d81-bd8e-7e59f6d63d03.png)

_After_
![Screen Shot 2022-12-14 at 9 26 58 AM](https://user-images.githubusercontent.com/11482842/207637961-4b24765d-8f07-4364-9ace-df282cb331a5.png)

**Migration Wizard**
_Before_
![Screen Shot 2022-12-14 at 9 25 07 AM](https://user-images.githubusercontent.com/11482842/207637995-3b3f4c7d-84a8-4c07-8d51-ddd4053c2ee8.png)

_After_
![Screen Shot 2022-12-14 at 9 25 29 AM](https://user-images.githubusercontent.com/11482842/207638026-8f765031-794b-47eb-9ad2-209304520f95.png)

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
